### PR TITLE
MODORDERS-254/MODORDERS-255 Rename assignments permissions

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
Unblock  [UIAC-4](https://issues.folio.org/browse/UIAC-4) 

## Approach
- renamed: 
  * orders.acquisitions-units-assignments.item.post->orders.acquisitions-units-assignments.assign
  * orders.acquisitions-units-assignments.item.delete->orders.acquisitions-units-assignments.manage
- acquisitions-units-assignments.assign required only in POST /orders/composite-orders if acqUnitIds not empty
- orders.acquisitions-units-assignments.manage required only in PUT /orders/composite-orders/id if acqUnitIds is changed
  

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [x] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [x] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java [hunk 1]
```diff
@@ -652,31 +655,41 @@ private boolean hasNewPoLines(CompositePurchaseOrder compPO, JsonArray poLinesFr
   }
 
   /**
-   * The method compares the permissions that are necessary to manage the acquisition units assignments and the permissions granted by the user,
-   * and if there is no required permission among the granted permissions, it throws {@link HttpException} with status 403.
+   * The method checks if user trying assign order to acquisition unit, if yes that
```
Comment (javalkar-varun) [inline src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java]: Going by what method is doing, think it should be something like `The method checks if the order is assigned to acquisition unit, if yes, then check that if the user has desired permission to assign the record to acquisition unit`

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -555,22 +556,22 @@
       ]
     },
     {
-      "permissionName": "orders.acquisitions-units-assignments.item.post",
+      "permissionName": "orders.acquisitions-units-assignments.assign",
       "displayName": "Acquisitions unit assignment - create unit assignment",
-      "description": "Create a new acquisitions unit assignments"
+      "description": "Assign new order to acquisitions units"
```
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json]: should this be assign acq units to new order

Comment (pautau-s) [reply descriptors/ModuleDescriptor-template.json]: I thought we were assigning an order to a unit, and not vice versa.

File: src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java (hunk 1 above)
Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java]: Updated

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (pautau-s) [reply descriptors/ModuleDescriptor-template.json]: @KVupp I'm going to merge PR to get the changes to testing env.  I will change the description in the following PR for MODORDERS-254/MODORDERS-255, if needed.

**Pull request merged**